helix-view/src/view.rs

```diff
@@ -176,84 +193,132 @@ impl View {
         &self,
         doc: &Document,
         scrolloff: usize,
-    ) -> Option<(usize, usize)> {
-        self.offset_coords_to_in_view_center(doc, scrolloff, false)
+    ) -> Option<ViewPosition> {
+        self.offset_coords_to_in_view_center::<false>(doc, scrolloff)
     }
 
-    pub fn offset_coords_to_in_view_center(
+    pub fn offset_coords_to_in_view_center<const CENTERING: bool>(
```

Comment:
@cessen also removed a similar use in ropey https://github.com/cessen/ropey/commit/633143d5ff435daf2c16ff57d092aab1438b88ac

Reply:
In this case I kept it might cause this function is quite performance sensitive and it's always k own at compiletime which version we want. So this is a performance optimization here

Reply:
I personally like the conicesness but if you guys don't like this for stylistic reasons I can change it to a bool parameter

Reply:
Note that in Ropey I removed the const generic parameter at least partly because I was pretty sure the code could be refactored to not need the parameter at all, which did indeed turn out to be the case.

Having said that, my personal feeling about const generic parameters is that if they can trivially be normal parameters (e.g. no other code would actually change), then they *probably* should be.  But if there are substantial measurable performance gains, I'm not actually against it.  It just feels semantically dirty to me.

Reply:
I kept the const generic for now as this function is called quite often and it's nice to ensure that these two functions are always distinct. While the performance impact isn't huge it's nice to keep latency down in hot functions like this. 

That being said if you guys prefer to keep this as a normal parameter for clarity reasons I wil happily change the signature.